### DISCUSSION TITLE
How to download PDF with audiobooks ?

### DISCUSSION THREAD

**Discussion by Rexadev** [audiamus/BookLibConnect#71] [General]
Some audiobooks have pdfs that show when added to library Example https://www.audible.in/pd/Industry-of-Anonymity-Audiobook/B07HZ466NC

**Comment by audiamus** [comment]
Associated PDF downloads are not yet supported, simply because so far I've not come across books that have them.

**Reply by audiamus** [reply]
Can't you download those PDFs with your web browser?

**Reply by Rexadev** [reply]
Would be nice if you can automate the process

**Reply by rbielaws** [reply]
I have at least a couple dozen books in my library that include a PDF so I could do some data collection and testing should you ever have the time and inclination to look at it.  

I definitely think it would be terribly convenient because you wouldn't need to keep track of which books have PDFs and deal with them in a separate workflow.  You could just expect to see it ... hmm, where?  In download only; in export too? ... Export would beg to be moved to your AAX Audio Converter Destination Folder.  Hmm.  It's probably a bigger undertaking than the OP realized before any incorporation of it here really makes things any simpler than just downloading it yourself to the location you want it.  Low priority for sure but terribly cool!

Anyway, should you decide, I'm a retired software designer and data architect so I'm capable of lending a hand.   It wasn't hard to find the URL in LicenseResponse.json (I removed some data but you'll recognize where to look if one exists).   My user ID resides at gmail.com if you need to connect.

    "drm_type": "Adrm",
    "granted_right": "Ownership",
    "message": "Ownership: User [xxxxxxxxxxxx] has [Purchase] rights.",
    "pdf_url": "https://d2fahduf2624mg.cloudfront.net/pre_purchase_docs/BK_NAXO_001984/2020-06-24-12-46-50/bk_naxo_001984.pdf",
    "requires_ad_supported_playback": false,
    "status_code": "Granted",

**Reply by audiamus** [reply]
If it is similar to downloading the cover image, by simply evaluating this `pdf_url` field, it shouldn't be much of a problem to integrate it. I put it on the todo list.

**Comment by rbielaws** [comment]
I'm up to 600 books.  Dealing with PDF's, despite being relatively rare,  still means over 100 for me.  And it's such a tedious process over time that I actually tried to add the feature myself.

Unfortunately the code structure doesn't appear to lend itself to the task.  The location of the PDF isn't available yet at the point where existing functionality for downloading executes.  Later when the code that retrieves the PDF location does execute, the code to  download is both tied to other structures, out of scope.  If I could get the PDF into the same location as the cover-art, I'm confident I could easily make changes needed to have it follow the cover art to the final converter output directory.

I know you have no real incentive to add this.  And, I really would have just created a pull request if I thought I understood the existing structure well enough to make the download possible.  But I'm not a GUI programmer, much less familiar with Windows. So I just can't follow the UI interaction well enough to understand the structural changes needed to make it happen without breaking who-knows-what.

Maybe it's easier than I think and you can tell me "All you need to do is ...."  In which case I'll try again.
Or, maybe even you think it's really to hard and I should stop thinking about it.  
I'm hoping you will share your thoughts.
